flatgeobuf.qmd

```diff
@@ -0,0 +1,108 @@
+
+    If you keep going, you'll quickly find which items
+
+    The [Wikipedia article](https://en.wikipedia.org/wiki/R-tree) and this [Mapbox blog post](https://blog.mapbox.com/a-dive-into-spatial-search-algorithms-ebd0c5e39d2a) are great resources for better understanding how R-Trees work.
+
+- _Hilbert_: The elements of an R-Tree must be _sorted_ before insertion to make the R-Tree useful. This is because the core benefit of an R-Tree is to _exclude_ elements that aren't within a spatial filter. If elements
+
+    But how do you sort geometries? They encompass two dimensions and a range of shapes. If you sort all geometries first on the `x` coordinate, then you may pair geometries that are far from each other on the `y` dimension. Instead, it's ideal to use a [_space-filling curve_](https://en.wikipedia.org/wiki/Space-filling_curve). That's math jargon, but essentially defines a way to group
+
+    A Hilbert R-Tree uses a [Hilbert Curve](https://en.wikipedia.org/wiki/Hilbert_curve), a special type of space-filling curve, to sort the centers of geometries. This ensures that geometries that are nearby on both the `x` _and_ `y` dimensions are placed close to each other in the R-Tree. This ensures that the resulting bounding boxes of the R-Tree are as small as possible, which means that the maximum number of elements can be discarded for any given spatial query.
+
+    Mathematically, a Hilbert Curve
+
+- _static_: FlatGeobuf files can't be modified without rewriting the entire file, so this [R-Tree](https://en.wikipedia.org/wiki/R-tree) is constructed in such a way that it can't be modified, which allows for improved tree generation.
+- _packed_: R-Trees work by having a series of leaves in the tree, where each leaf contains a number of inner leaves and so on. So that searching the tree doesn't take too much time, each leaf has a maximum number of items, usually something like 16. When R-Trees are intended to be mutable, many leaves contain _less than_ the maximum number of children, so that a child can be added without having to rebalance the tree.
+
+    Because we require this index to be static, we can make the R-Tree more efficient! We can create the minimum number of nodes to handle the known number of geometries, as we know there will be no added features later.
+
+    FlatGeobuf does this
+
+This index
+
+1. Maps every bounding box to a byte range within the file
+2. Allows for discarding features outside of a spatial query, having only the index.
+
+<!-- Keep in mind that since FlatGeobuf isn't chunked, -->
+
+And because FlatGeobuf is a row-based format, if some properties are really large, like if a column is a string description with many characters, you'll have a tradeoff between fetching lots of extra data to combine multiple requests into one or making many requests for small features that are scattered throughout the file.
+
+### Streaming features is supported
```

Comment:
```suggestion
### Streaming features are supported
```

Reply:
I was _thinking_ here `is` makes more sense than `are` because `supported` is describing `Streaming`, not `features`, the same as if it were just "Streaming is supported". But then again English was not my favorite class 😅

Reply:
you're right actually, I just wasn't reading it that way